openedx/features/wikimedia_features/auth_backend.py

```diff
@@ -0,0 +1,45 @@
+import logging
+from django.utils.functional import cached_property
+from social_core.exceptions import AuthException
+from common.djangoapps.third_party_auth.identityserver3 import IdentityServer3
+
+log = logging.getLogger(__name__)
+
+
+class WikimediaIdentityServer(IdentityServer3):
+    """
+    An extension of the IdentityServer3 for use with Wikimedia's IdP service.
+    """
+    name = "wikimediaIdentityServer"
+    DEFAULT_SCOPE = ["mwoauth-authonlyprivate"]
+    ID_KEY = "sub"
+
+    def get_user_details(self, response):
+        """
+        Returns detail about the user account from the service
+        """
+
+        try:
+            if response["realname"]:
+                fullname = response["realname"]
+                firstname, lastname = fullname.split(' ')
```

Comment:
this will break if the full name consists of three parts. e.g "aa bb cc"

Reply:
nice catch!, done.